File: fitness_app/backend/calorie_calculator.py
```python
def calculate_bmr(weight, height, age, sex):
    if sex == 'male':
        bmr = 88.362 + (13.397 * weight) + (4.799 * height) - (5.677 * age)
    else:
        bmr = 447.593 + (9.247 * weight) + (3.098 * height) - (4.330 * age)
    return bmr

def calculate_tdee(bmr, activity_level):
    if activity_level == 'sedentary':
        tdee = bmr * 1.2
    elif activity_level == 'light':
        tdee = bmr * 1.375
    elif activity_level == 'moderate':
        tdee = bmr * 1.55
    elif activity_level == 'active':
        tdee = bmr * 1.725
    else:
        tdee = bmr * 1.9
    return tdee

def calculate_calorie_goal(tdee, weight_goal):
    if weight_goal == 'lose':
        calorie_goal = tdee - 500
    elif weight_goal == 'gain':
        calorie_goal = tdee + 500
    else:
        calorie_goal = tdee
    return calorie_goal
```

File: fitness_app/frontend/src/components/CalorieCalculator.js
import React, { useState, useEffect } from 'react';
import axios from 'axios';

const CalorieCalculator = () => {
    const [sex, setSex] = useState('');
    const [height, setHeight] = useState('');
    const [weight, setWeight] = useState('');
    const [age, setAge] = useState('');
    const [activityLevel, setActivityLevel] = useState('');
    const [weightLossGoal, setWeightLossGoal] = useState('');
    const [calories, setCalories] = useState('');

    useEffect(() => {
        if (sex && height && weight && age && activityLevel && weightLossGoal) {
            axios.post('/api/calorie_calculator', {
                sex,
                height,
                weight,
                age,
                activityLevel,
                weightLossGoal
            })
            .then(response => {
                setCalories(response.data.calories);
            })
            .catch(error => {
                console.error('Error calculating calories', error);
            });
        }
    }, [sex, height, weight, age, activityLevel, weightLossGoal]);

    return (
        <div className="calorieCalculator">
            <h2>Calorie Calculator</h2>
            <form>
                <label>
                    Sex:
                    <input type="text" value={sex} onChange={e => setSex(e.target.value)} />
                </label>
                <label>
                    Height (in cm):
                    <input type="number" value={height} onChange={e => setHeight(e.target.value)} />
                </label>
                <label>
                    Weight (in kg):
                    <input type="number" value={weight} onChange={e => setWeight(e.target.value)} />
                </label>
                <label>
                    Age:
                    <input type="number" value={age} onChange={e => setAge(e.target.value)} />
                </label>
                <label>
                    Activity Level:
                    <select value={activityLevel} onChange={e => setActivityLevel(e.target.value)}>
                        <option value="">--Please choose an option--</option>
                        <option value="sedentary">Sedentary</option>
                        <option value="light">Light</option>
                        <option value="moderate">Moderate</option>
                        <option value="active">Active</option>
                        <option value="veryActive">Very Active</option>
                    </select>
                </label>
                <label>
                    Weight Loss Goal (in kg):
                    <input type="number" value={weightLossGoal} onChange={e => setWeightLossGoal(e.target.value)} />
                </label>
            </form>
            <div>
                <h3>Your daily calorie intake should be: {calories}</h3>
            </div>
        </div>
    );
};

export default CalorieCalculator;